src/controls/EnumControl.tsx

```diff
@@ -0,0 +1,100 @@
+import type { ControlProps as JSFControlProps } from "@jsonforms/core"
+import { Col, Form, Select, Segmented, Radio } from "antd"
+import type { Rule } from "antd/es/form"
+import { EnumControlOptions, ControlUISchema } from "../ui-schema"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+
+type ControlProps = Omit<JSFControlProps, "uischema"> & {
+  uischema: ControlUISchema<unknown> | JSFControlProps["uischema"]
+}
+
+export const EnumControl = (props: ControlProps) => {
+  if (!props.visible) return null
+
+  const rules: Rule[] = [
+    { required: props.required, message: `${props.label} is required` },
+  ]
+
+  const formItemProps =
+    "formItemProps" in props.uischema ? props.uischema.formItemProps : {}
+
+  const defaultValue =
+    (props.data as unknown) ?? (props.schema.default as unknown)
+
+  const appliedUiSchemaOptions = props.uischema.options as EnumControlOptions
+
+  const enumValueToLabelMap = appliedUiSchemaOptions?.enumValueToLabelMap
+  const options = props.schema.enum
+    ? props.schema.enum?.map((value) => ({
+        label:
+          enumValueToLabelMap && typeof value === "string"
+            ? enumValueToLabelMap[value]
+            : (value as string),
+        value: value as string,
```

Comment:
why do you have to cast as a string here?

Reply:
Because [enum](https://jsonforms.io/api/core/interfaces/jsonschema7#enum) is an `any`